### PR TITLE
Add detailed docstrings and usage examples to Ersilia'a serve modules

### PR DESCRIPTION
**Pull request by Abellegese**
Thank you for taking your time to contribute to Ersilia, just a few checks before we proceed
- [ ] Have you followed the guidelines in our [Contribution Guide](https://github.com/ersilia-os/ersilia/blob/master/CONTRIBUTING.md)
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

**Description**

This PR improves the documentation of several serve module files by adding detailed docstrings, summaries, and usage examples to public functions.

### CODE REVIEW

File: ersilia/serve/api.py [hunk 1]
```diff
@@ -114,12 +137,66 @@ def _post(self, input, output):
             result = json.dumps(result, indent=4)
             result = self.__result_returner(result, output)
         if result is None and len(input) == 1:
-            # if the only one prediction did not work, return empty
             result = [{"input": input[0], "output": self._empty_output}]
             result = json.dumps(result, indent=4)
```
Comment (DhanshreeA) [inline ersilia/serve/api.py:141]: Can we take this opportunity to take out this piece of code from the if blocks and just always do a Json dump and result returner?

**Review by DhanshreeA — CHANGES_REQUESTED** [review]

File: ersilia/serve/api.py (hunk 1 above)
Comment (Abellegese) [reply ersilia/serve/api.py:141]: OKay @DhanshreeA.

Comment (Abellegese) [reply ersilia/serve/api.py]: Hey @DhanshreeA I am getting `EmptyOutputError` after the change.

Comment (Abellegese) [reply ersilia/serve/api.py:141]: I put it back and it works. I think `output: None` can't be processed after its get generated.

**Pull request merged**